src/lib.rs

```diff
@@ -1601,6 +1602,40 @@ pub trait Itertools : Iterator {
         self.next().map(move |x| self.fold(x, f))
     }
 
+    /// Accumulate the elements in the iterator in a balanced manner.
+    ///
+    /// This produces a result like `f(f(1, 2), f(3, 4))`, a balanced call tree,
+    /// instead of `f(f(f(1, 2), 3), 4)`, a linear sequence.
```

Comment:
If we include a fifth argument here, we cover the asymmetry too, with `f(f(f(1, 2), f(3, 4)), 5)`, if that is readable?

Maybe we can do an ascii art that draws `f(f(f(1, 2), f(3, 4)), 5)` as a tree

Reply:
Not that pretty, but why not

```text
        f
       / \
      /   \
     f     5
    / \
   /   \
  f     f
 / \   / \
1   2 3   4
```

Reply:
Or maybe tilted?  I think of it this way:
```
        f
       /|
      f |
    _/| |
   /  | |
  f   f |
 /|  /| |
1 2 3 4 5
```
(Well, I actually think of it as that but upside down, so it's in dataflow order.)

Reply:
I think there's too little air in the last version so I like the first one more -- easier to parse for me.

Reply:
Or one that can be read left-to-right, top-to-bottom:
```text
1
 \
2-f
   \
3   \
 \   \
4-f---f
       \
5-------f
```

Reply:
Hmm, maybe an orthogonal graph using box drawings:
```text
1 2 3 4 5 6 7
│ │ │ │ │ │ │
└─f └─f └─f │
  │   │   │ │
  └───f   └─f
      │     │
      └─────f
```

Reply:
oh wow, I just wanted to draw a diagram not start an ascii-off :smile:

Box drawings look nice in rustdoc, so that's fine too. The last one is good.